src/session/sidebar/mini_thumbnail.rs

```diff
@@ -0,0 +1,99 @@
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gdk, glib, graphene};
+
+mod imp {
+    use super::*;
+    use once_cell::sync::Lazy;
+    use std::cell::RefCell;
+
+    #[derive(Debug, Default)]
+    pub(crate) struct MiniThumbnail(pub(super) RefCell<Option<gdk::Paintable>>);
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for MiniThumbnail {
+        const NAME: &'static str = "SidebarMiniThumbnail";
+        type Type = super::MiniThumbnail;
+        type ParentType = gtk::Widget;
+    }
+
+    impl ObjectImpl for MiniThumbnail {
+        fn properties() -> &'static [glib::ParamSpec] {
+            static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+                vec![glib::ParamSpecObject::new(
+                    "paintable",
+                    "Paintable",
+                    "The paintable of the media",
+                    gdk::Paintable::static_type(),
+                    glib::ParamFlags::READWRITE | glib::ParamFlags::EXPLICIT_NOTIFY,
+                )]
+            });
+            PROPERTIES.as_ref()
+        }
+
+        fn set_property(
+            &self,
+            obj: &Self::Type,
+            _id: usize,
+            value: &glib::Value,
+            pspec: &glib::ParamSpec,
+        ) {
+            match pspec.name() {
+                "paintable" => obj.set_paintable(value.get().unwrap()),
+                _ => unimplemented!(),
+            }
+        }
+
+        fn property(&self, obj: &Self::Type, _id: usize, pspec: &glib::ParamSpec) -> glib::Value {
+            match pspec.name() {
+                "paintable" => obj.paintable().to_value(),
+                _ => unimplemented!(),
+            }
+        }
+    }
+
+    impl WidgetImpl for MiniThumbnail {
+        fn snapshot(&self, widget: &Self::Type, snapshot: &gtk::Snapshot) {
+            if let Some(paintable) = self.0.borrow().as_ref() {
+                snapshot.translate(&graphene::Point::new(0.0, 0.0));
+                paintable.snapshot(
+                    snapshot.upcast_ref(),
+                    widget.width() as f64,
+                    widget.height() as f64,
+                );
```

Comment:
This way the image will appear stretched when it's not perfectly squared. For instance (Telegrand on the left):

![image](https://user-images.githubusercontent.com/23530586/172501802-553bbaa4-5958-44d4-8ada-3780e994b6be.png)

This is basically `object-fit: fill`, but we need `cover`. The snapshot implementation in `MediaPicture` basically does that, so you can just use that.